web/src/settings_bots.ts

```diff
@@ -324,13 +361,17 @@ export function add_a_new_bot() {
     });
 }
 
-export function set_up() {
+export function set_up(): void {
     $("#download_botserverrc").on("click", function () {
         let content = "";
 
         for (const bot of bot_data.get_all_bots_for_current_user()) {
             if (bot.is_active && bot.bot_type === OUTGOING_WEBHOOK_BOT_TYPE_INT) {
-                const bot_token = bot_data.get_services(bot.user_id)[0].token;
+                const services = bot_data.get_services(bot.user_id);
+                assert(services !== undefined);
+                const service = services[0];
+                assert(service && "token" in service);
```

Comment:
Not the best practise to `"token" in service` ,but I guess we don't have another option.

Reply:
yes, i don't think there is any other way for this.

Reply:
The downside of the `in` is the lack of ts compiler warnings. But since this is used in a runtime assert there are no drawbacks.